Automattic/wp-calypso
# Command Palette: Only fetch sites in Calypso

## Proposed Changes

Prevents the Command Palette from making a failing and unnecessary API request to the `/me/sites` endpoint in WP Admin:

- The Command Palette component now has a `useSites` property so consumers can decide if and how the sites list for multi-site commands are fetched.
- Since we only support multi-site commands in Calypso, only the Calypso command palette sets a `useSites` prop which is mapped to a new `useSiteExcerptsSorted` hook.
- The new `useSiteExcerptsSorted` hook fetches the sites [just like the old command palette component](https://github.com/Automattic/wp-calypso/blob/5bc167f74783b600b115732e8a2a08683e4f7641/client/components/command-palette/use-command-palette.tsx#L134-L145).
- For similar reasons, add a new `userCapabilities` property so consumers can decide how the user capabilities are fetched.
- In Calypso, capabilities are populated [the same way we did before](https://github.com/Automattic/wp-calypso/blob/5bc167f74783b600b115732e8a2a08683e4f7641/client/components/command-palette/use-command-palette.tsx#L157).

This allowed to remove a bunch of unnecessary code.

## Testing Instructions

- Use the Calypso live link below
- Make sure multi-site commands continue to work in Calypso
- Checkout this PR
- Build the `@automattic/command-palette` package:
  - `yarn distclean`
  - `yarn install`
- Build and sync to your sandbox the `@automattic/command-palette-wp-admin` app:
  - `cd apps/command-palette-wp-admin`
  - `yarn dev --sync`
- Apply https://github.com/Automattic/jetpack/pull/35635 to your sandbox
- Sandbox `widgets.wp.com`
- Load WP Admin on your sandboxed site and press `⌘K`
- Make sure the command palette does not make any API request to the `/me/sites` endpoint.
